### PR TITLE
fix: kubesphere/api package UT failed

### PR DESCRIPTION
**Pull request by RolandMa1986**
Signed-off-by: Roland.Ma <rolandma@yunify.com>

### What type of PR is this?
/kind failing-test

### What this PR does / why we need it:
UT was failed when running on kubesphere/api repo. Because kube-openapi and kube-apiserver packages don't compatible. 

### Which issue(s) this PR fixes:
Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

/cc @yuswift

### CODE REVIEW

**Comment by yuswift** [comment]
/lgtm
BTW, please take a look at the failing e2e test.

File: Makefile [hunk 1]
```diff
@@ -139,6 +139,8 @@ helm-uninstall: ; $(info $(M)...Begin to helm-uninstall.)  @ ## Helm-uninstall.
 # Run tests
 test: vet ;$(info $(M)...Begin to run tests.)  @ ## Run tests.
 	export KUBEBUILDER_CONTROLPLANE_START_TIMEOUT=2m; go test ./pkg/... ./cmd/... -covermode=atomic -coverprofile=coverage.txt
+	cd staging/src/kubesphere.io/api ; GOFLAGS="" go test ./...
+	cd staging/src/kubesphere.io/client-go ; GOFLAGS="" go test ./...
```
Comment (wansir) [inline Makefile:143]: I think the logic is similar to: https://github.com/kubesphere/kubesphere/blob/master/hack/update-vendor.sh#L124-L134

Comment (RolandMa1986) [reply Makefile:143]: Do you mean to refactor the commands with a shell?

Comment (wansir) [reply Makefile:143]: Another staging library may be added, should not be hard code here.

Comment (yuswift) [reply Makefile:143]: Could we merge this pr first? We can refactor the commands with another pr. Some work has been blocked because of this pr.

Comment (zryfish) [reply Makefile:143]: Any more comments? @wansir

**Pull request merged**